src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs

```diff
@@ -41,6 +42,15 @@ public async Task<bool> HandleAsync(AddAatf message)
 
             Operator @operator = await dataAccess.GetSingleByExpression<Operator>(new OperatorByOrganisationIdSpecification(message.OrganisationId));
 
+            if (@operator == null)
+            {
+                Organisation organisation = await dataAccess.GetById<Organisation>(message.OrganisationId);
+
+                @operator = new Operator(organisation);
+
+                await dataAccess.Add<Operator>(@operator);
+            }
+
             AatfContact contact = contactMapper.Map(message.AatfContact);
 
             Aatf aatf = new Aatf(
```

Comment:
In the unit test can we check that the AATF is added to the context with all of the matching arguments... at the moment its currently only name

Reply:
Done